library/core/src/ptr/mod.rs

```diff
@@ -2388,4 +2404,51 @@ mod verify {
         let copy = unsafe { read_volatile(ptr) };
         assert_eq!(val, copy);
     }
+
+    fn check_align_offset<T>(p: *const T) {
```

Comment:
For my own edification--why did I not need a `#[kani::unwind]` annotation here? I thought the unbounded loop inside `mod_inv` would mean Kani would need one to symbolically execute `align_offset`.

Reply:
Yes, this seems surprising. You'd want to look at coverage information to see whether you reach this code at all. Doing so might be somewhat cumbersome across several harnesses, so you could approximate it by inserting `assert!(0);` into the code (perhaps multiple of them at once) to see which lines of code you actually reach.

Reply:
Good suggestion, thanks. Seems like I'm getting coverage in all of the code that `align_offset` calls except for the body of the `loop`. I'm a bit confused why Kani would just skip it entirely--I would have expected it to try to execute it and then hang, but not sure how to address the lack of coverage. Any ideas?

Reply:
Are you getting beyond the loop?

Reply:
Yes. Full coverage results from running 
`kani verify-std -Z unstable-options ./library --target-dir /tmp/verify-rust-std -Z function-contracts -Z mem-predicates -Z line-coverage --coverage --harness check_align_offset_u64` below:

<details>
  <summary>Line Coverage</summary>
  Coverage Results:

  /Users/cmzech/verify-std/library/core/src/lib.rs, 0, PARTIAL
  /Users/cmzech/verify-std/library/core/src/lib.rs, 428, PARTIAL

/Users/cmzech/verify-std/library/core/src/mem/mod.rs, 311, FULL
/Users/cmzech/verify-std/library/core/src/mem/mod.rs, 312, FULL

/Users/cmzech/verify-std/library/core/src/num/mod.rs, 1259, FULL

/Users/cmzech/verify-std/library/core/src/ops/function.rs, 250, PARTIAL

/Users/cmzech/verify-std/library/core/src/ptr/const_ptr.rs, 61, FULL
/Users/cmzech/verify-std/library/core/src/ptr/const_ptr.rs, 62, FULL
/Users/cmzech/verify-std/library/core/src/ptr/const_ptr.rs, 145, FULL
/Users/cmzech/verify-std/library/core/src/ptr/const_ptr.rs, 146, FULL

/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1887, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1889, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1890, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1892, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1894, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1896, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1897, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1898, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1899, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1900, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1918, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1930, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1931, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1932, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1942, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1949, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1951, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1952, NONE
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1953, NONE
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1954, NONE
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1957, NONE
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1959, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1960, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1962, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1968, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1971, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1973, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1977, NONE
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1978, NONE
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1982, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1983, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 1999, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2000, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2005, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2008, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2010, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2013, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2015, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2027, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2028, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2029, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2030, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2033, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2035, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2059, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2062, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2065, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2069, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2072, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2076, PARTIAL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2077, NONE
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2412, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2413, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2414, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2440, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2442, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2443, FULL
/Users/cmzech/verify-std/library/core/src/ptr/mod.rs, 2444, FULL

Verification Time: 5.6330814s

Complete - 1 successfully verified harnesses, 0 failures, 1 total.

</details>